### PR TITLE
N8N-3750 redesign button component

### PR DESCRIPTION
**Pull request by alexgrozav**
- Button component redesign
   - new stories (+ typescript)
   - implementation update (improved styles reuse, reduced size)
   - active, hover, focus states
   - changed `type` prop to be used for colors and added `outline` and `text` as modifier props
   - refactored usages in editor-ui
  
- Design system update
   - added neutral color
   - generated color shades from 50 to 950 (in increments of 50) for primary, secondary, neutral, success, warning, danger

### CODE REVIEW

File: packages/design-system/src/components/N8nButton/Button.vue [hunk 1]
```diff
@@ -68,11 +66,19 @@ export default {
...
 		},
+		outline: {
+			type: Boolean,
+			default: false,
+		},
+		text: {
+			type: Boolean,
+			default: false,
+		},
 		icon: {
```
Comment (MiloradFilipovic) [inline packages/design-system/src/components/N8nButton/Button.vue:77]: Should we have a `type` for `icon`?

File: packages/design-system/theme/src/button.scss [hunk 2]
```diff
@@ -28,17 +27,29 @@ $loading-overlay-background-color: rgba(255, 255, 255, 0.35);
...
 	&:active {
 		color: var.$button-active-color;
 		border-color: var.$button-active-border-color;
 		background-color: var.$button-active-background-color;
 		outline: none;
 	}
 
+  	&:focus {
+	  	border-color: var.$button-focus-outline-color;
+	    outline: 2px solid var.$button-focus-outline-color;
```
Comment (MiloradFilipovic) [inline packages/design-system/theme/src/button.scss:50]: Would it be beneficial to extract this outline width as a variable to we can use it for other components focus style?

**Review by MiloradFilipovic — APPROVED** [review]
Looks good. I left some minor comments inline.
One more nitpick (not related to this PR): Should welcome sticky thumbnail be updated to use new buttons:
image

File: packages/design-system/theme/src/button.scss (hunk 2 above)
Comment (alexgrozav) [reply packages/design-system/theme/src/button.scss]: Great suggestion. Done! :)

File: packages/design-system/src/components/N8nButton/Button.vue (hunk 1 above)
Comment (alexgrozav) [reply packages/design-system/src/components/N8nButton/Button.vue:77]: I missed that. Added type `String`.

**Comment by alexgrozav** [comment]
> Looks good. I left some minor comments inline.
> One more nitpick (not related to this PR): Should welcome sticky thumbnail be updated to use new buttons:
> image

Will discuss this with Max. We should at least replace the thumbnail button.

File: packages/design-system/src/components/N8nButton/Button.stories.ts [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+	argTypes: {
+		label: {
+			control: 'text',
+		},
+		title: {
+			control: 'text',
+		},
+		type: {
+			control: 'select',
+			options: ['primary', 'outline', 'light', 'text', 'tertiary'],
```
Comment (mutdmour) [inline packages/design-system/src/components/N8nButton/Button.stories.ts:18]: need to update the types here to match new ones

File: packages/design-system/src/components/N8nButton/Button.vue [hunk 4]
```diff
@@ -68,11 +66,20 @@ export default {
...
+			type: String,
 		},
 		round: {
 			type: Boolean,
-			default: true,
+			default: false,
 		},
 		circle: {
```
Comment (mutdmour) [inline packages/design-system/src/components/N8nButton/Button.vue:84]: maybe we can delete these two... based on a quick search, we are only using `circle` in `icon button` and that needs to be changed... also `round` does not seem to be used anymore

File: packages/design-system/src/components/N8nIconButton/IconButton.vue [hunk 5]
```diff
@@ -7,6 +7,8 @@
 		:title="props.title"
 		:icon="props.icon"
 		:theme="props.theme"
+	    :outline="props.outline"
+	    :text="props.text"
```
Comment (mutdmour) [inline packages/design-system/src/components/N8nIconButton/IconButton.vue:11]: nitpick - fix spacing please

File: packages/design-system/theme/src/_tokens.scss [hunk 6]
```diff
@@ -173,6 +175,15 @@
 		var(--color-info-tint-2-l)
 	);
 
+	--color-neutral-h: 220;
```
Comment (mutdmour) [inline packages/design-system/theme/src/_tokens.scss:178]: can you add a demo page for this?

File: packages/design-system/theme/src/_tokens.scss [hunk 7]
```diff
@@ -355,6 +366,15 @@
 		var(--color-sticky-default-border-l)
 	);
 
+  	--color-white: hsl(0, 0, 100, 1);
+
+	@each $color in ('primary', 'neutral', 'secondary', 'success', 'warning', 'danger') {
```
Comment (mutdmour) [inline packages/design-system/theme/src/_tokens.scss:371]: would be nice if we could demo these as well in demo pages

**Review by mutdmour — COMMENTED** [review]
awesome job.. though I think icon buttons need to be updated too to match.

File: packages/design-system/theme/src/_tokens.scss [hunk 8]
```diff
@@ -355,6 +366,15 @@
 		var(--color-sticky-default-border-l)
 	);
 
+  	--color-white: hsl(0, 0, 100, 1);
```
[the remark this answers is not in the document]
Comment (mutdmour) [reply packages/design-system/theme/src/_tokens.scss:369]: FYI there's already an xlight background and foreground colors we are using... goal there was to allow theming to work easily...

File: packages/design-system/src/components/N8nButton/Button.stories.ts (hunk 3 above)
Comment (alexgrozav) [reply packages/design-system/src/components/N8nButton/Button.stories.ts]: Good catch. Updated!

File: packages/design-system/src/components/N8nButton/Button.vue (hunk 4 above)
Comment (alexgrozav) [reply packages/design-system/src/components/N8nButton/Button.vue]: Good point. Done! :)

File: packages/design-system/src/components/N8nIconButton/IconButton.vue (hunk 5 above)
Comment (alexgrozav) [reply packages/design-system/src/components/N8nIconButton/IconButton.vue]: Fixed

File: packages/design-system/theme/src/_tokens.scss (hunk 6 above)
Comment (alexgrozav) [reply packages/design-system/theme/src/_tokens.scss]: Added a story for it.

File: packages/design-system/theme/src/_tokens.scss (hunk 7 above)
Comment (alexgrozav) [reply packages/design-system/theme/src/_tokens.scss]: These are still up for discussion. Max said we'll add numbered shades in a later iteration. Should I comment them out?

File: packages/design-system/theme/src/_tokens.scss (hunk 8 above)
[the remark this answers is not in the document]
Comment (alexgrozav) [reply packages/design-system/theme/src/_tokens.scss]: Added color white story. The design in Figma has it defined, so I just tried to align the two.

File: packages/design-system/theme/src/utilities/_float.scss [hunk 9]
```diff
@@ -0,0 +1,7 @@
+.float-left {
+  float: left !important;
+}
+
+.float-right {
+  float: right !important;
+}
```
Comment (mutdmour) [inline packages/design-system/theme/src/utilities/_float.scss:7]: don't seem to be used anywhere. might be nice if they are documented in storybook like other utilities function.. (you don't have to put in too much effort there)

File: packages/design-system/theme/src/_tokens.scss (hunk 7 above)
Comment (mutdmour) [reply packages/design-system/theme/src/_tokens.scss]: ya if we are not using them anywhere and Max is not sold, let's comment them out or delete them.

File: packages/design-system/theme/src/_tokens.scss (hunk 8 above)
[the remark this answers is not in the document]
Comment (mutdmour) [reply packages/design-system/theme/src/_tokens.scss]: ya the thing about figma is that it needs to be updated to match our tokens... it's tech debt for Max. I would recommend we remove it and stick to using foreground/background ones unless there's a clear usecase.

File: packages/design-system/src/components/N8nButton/Button.stories.ts [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+				type: 'boolean',
+			},
+		},
+		icon: {
+			control: {
+				type: 'text',
+			},
+		},
+		fullWidth: {
+			type: 'boolean',
```
Comment (mutdmour) [inline packages/design-system/src/components/N8nButton/Button.stories.ts:47]: can be removed or replaced with block

File: packages/design-system/src/components/N8nButton/Button.vue [hunk 11]
```diff
@@ -68,62 +56,114 @@ export default {
 			type: Boolean,
 			default: false,
 		},
-		icon: {
-		},
-		round: {
+		active: {
```
Comment (mutdmour) [inline packages/design-system/src/components/N8nButton/Button.vue:59]: why do we need an active prop? is this used anywhere? I thought active was more of a state when you press a button.

File: packages/design-system/src/styleguide/colors.stories.mdx [hunk 12]
```diff
@@ -91,6 +91,32 @@ import ColorCircles from './ColorCircles.vue';
...
+		}}
+	</Story>
+</Canvas>
+
+## White
+
+<Canvas>
+	<Story name="white">
+		{{
+			template: `<color-circles :colors="['--color-white']" />`,
```
Comment (mutdmour) [inline packages/design-system/src/styleguide/colors.stories.mdx:112]: unless there's a usecase I don't understand here, would rather we stick to foreground and background xlight, so we could continue to separate between those usecases... Plan is for figma designs to follow this..

File: packages/design-system/src/components/N8nButton/Button.stories.ts (hunk 10 above)
Comment (mutdmour) [reply packages/design-system/src/components/N8nButton/Button.stories.ts:47]: in fact, you can probably delete most of the props here... storybook automatically adds props that are specified.. (it did not do that when I wrote this component initially)

File: packages/design-system/theme/src/utilities/_float.scss (hunk 9 above)
Comment (alexgrozav) [reply packages/design-system/theme/src/utilities/_float.scss:7]: Added stories for it.

File: packages/design-system/src/styleguide/colors.stories.mdx (hunk 12 above)
Comment (alexgrozav) [reply packages/design-system/src/styleguide/colors.stories.mdx]: Thank you for letting me know. Removed. Updated to background xlight.

File: packages/design-system/src/components/N8nButton/Button.vue (hunk 11 above)
Comment (alexgrozav) [reply packages/design-system/src/components/N8nButton/Button.vue]: Removed. Was going to add it as a separate explicit state, but we probably won't use this.

File: packages/design-system/src/components/N8nButton/Button.stories.ts (hunk 10 above)
Comment (alexgrozav) [reply packages/design-system/src/components/N8nButton/Button.stories.ts]: Good point! Removed props that are auto-generated.

**Pull request merged**